pennylane/operation.py

```diff
@@ -959,6 +959,30 @@ def compute_decomposition(*params, wires=None, **hyperparameters):
         """
         raise DecompositionUndefinedError
 
+    # pylint: disable=no-self-argument, comparison-with-callable
+    @classproperty
+    def has_decomposition_threshold(cls):
+        r"""Bool: Whether or not the Operator offers a threshold from computing the full matrix and
+        its decomposition as a product of other operators.
+
+        Note: Child classes may have this as an instance property instead of as a class property.
+        """
+        return (
+            cls.has_matrix
+            and cls.has_decomposition
+            and (cls.decomposition_threshold != Operator.decomposition_threshold)
+        )
+
+    @property
+    def decomposition_threshold(self):
```

Comment:
Would this decomposition threshold vary depending on the device? Or would this hold for all simulators? If it's simulator-dependent, I would say this is information that the device would need to store, instead of the operation.

Even if it makes sense to be stored by the device, the only operators where this matters (so far) are `Permute` and `QFT`.  Maybe instead, it makes sense for the stopping condition to use `getattr(op, "decomposition_threshold", None)` and only define it in relevant classes.

Reply:
Why not generalising this to any operator?

For example, we could add a `use_decomposition` property to the needed operators that returns `True` if the number of wires used is higher than the `"decomposition_threshold"`. This way in the device we can just do:
```python
if getattr(obj, "use_decomposition", False):
    return obj.use_decomposition
```

Reply:
Hi @AlbertMitjans, what you say makes sense.
The approach you propose is very similar to the first direction I follow.
But we decide to keep the decision fully owned by the device, and not as a property of the operation (in PL core).
For example on `lightning.qubit`, we may decide to choose different thresholds or even not use it at all, for some cases.
Since we only have a couple (2) cases, we decide to hard-code it for now.
There are no plans to expand this list in the near future.
Maybe @albi3ro would like to expand on that.